server/server/Controllers/AccountController.cs

```diff
@@ -0,0 +1,104 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using server.Models;
+using System.Security.Claims;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.Cookies;
+using Microsoft.EntityFrameworkCore;
+using server.DBRepositories;
+
+namespace server.Controllers
+{
+    public class AccountController : Controller
+    {
+        private UnitOfWork db;
+        public AccountController(ApplicationContext context)
+        {
+              db = new UnitOfWork(context);
+        }
+
+        [Route("/role")]
+        [HttpGet]
+        public string GetRole()
+        {
+            string userEmail = HttpContext.User.Identity.Name;
+            User user = db.Users.GetByEmail(userEmail);//.FirstOrDefault(u => u.Email == userEmail);
+            if(user != null)
+            {
+                return AuthResponse.User;
+            }
+
+            Admin admin = db.Admins.GetByEmail(userEmail);//.FirstOrDefault(u => u.Email == userEmail);
+            if(admin != null)
+            {
+                return AuthResponse.Admin;
+            }
+                
+            return AuthResponse.Guest;
+        }
+
+        [Route("login")]
+        [HttpPost]
+        public async Task<IActionResult> Login(Login model)
+        {
+            User user = db.Users.GetAll().FirstOrDefault(u => u.Email == model.Email && u.Password == model.Password);
```

Comment:
GetAll is useless, because now all users are read to memory and then are filtered. Better to use FirstOrDefault to filter users in database